contracts/ApprovalWorkflow.sol

```diff
@@ -1,53 +1,102 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.0;
 
-import "./BidEvaluation.sol"; // Import the BidEvaluation contract
+import "./BidEvaluation.sol"; // Import the BidEvaluation contract if it's in a separate file
 
 contract ApprovalWorkflow {
     BidEvaluation bidEvaluationContract; // Instance of the BidEvaluation contract
-
+    address public owner;
+    mapping(address => bool) public whitelist; // Tracks whitelisted officer addresses
+    
     enum ApprovalStatus { Pending, Approved, Rejected }
     struct BidApproval {
         uint256 rfpId;
         address vendor;
         ApprovalStatus status;
     }
-
+    
+    struct ApprovalData {
+        uint256 approvalCount;
+        mapping(address => bool) approvers;
+        bool isApproved;
+        bool isRejected;
+    }
+    
     mapping(uint256 => BidApproval) public approvals;
+    mapping(uint256 => ApprovalData) private approvalData;
+    
+    uint256 public constant MIN_APPROVALS = 2;
 
     event BidApproved(uint256 indexed rfpId, address indexed vendor);
     event BidRejected(uint256 indexed rfpId, address indexed vendor);
+    event OfficerAddedToWhitelist(address officer);
+    event OfficerRemovedFromWhitelist(address officer);
 
     constructor(address _bidEvaluationContractAddress) {
         bidEvaluationContract = BidEvaluation(_bidEvaluationContractAddress);
+        owner = msg.sender; // Setting the deployer as the owner
     }
 
-    // Function to approve a bid
-    function approveBid(uint256 _rfpId, address _vendor) public {
-        require(bidEvaluationContract.isWinningBid(_rfpId, _vendor), "Vendor is not the winning bidder");
+    modifier onlyWhitelisted() {
+        require(whitelist[msg.sender], "Caller is not whitelisted");
+        _;
+    }
+
+    modifier onlyOwner() {
+        require(msg.sender == owner, "Caller is not the owner");
+        _;
+    }
 
-        approvals[_rfpId] = BidApproval({
-            rfpId: _rfpId,
-            vendor: _vendor,
-            status: ApprovalStatus.Approved
-        });
+    modifier notAlreadyApproved(uint256 _rfpId) {
+        require(!approvalData[_rfpId].approvers[msg.sender], "Officer has already approved this bid");
+        _;
+    }
+
+    function addToWhitelist(address _officer) public onlyOwner {
+        whitelist[_officer] = true;
+        emit OfficerAddedToWhitelist(_officer);
+    }
 
-        emit BidApproved(_rfpId, _vendor);
+    function removeFromWhitelist(address _officer) public onlyOwner {
+        whitelist[_officer] = false;
+        emit OfficerRemovedFromWhitelist(_officer);
```

Comment:
- Same here as above, do we want to use OpenZeppelins's Acccess Control? Although I do think we might not have enough time so this should suffice.

Reply:
Updated with Openzeppelin Contract